### ISSUE TITLE
Possibility to disable QueryAnalyzer in settings

### ISSUE THREAD

**Issue by stulnikov-roud** [wp-graphql/wp-graphql#2535]
### What problem does this address?

QueryAnalyzer adds `X-GraphQL-*` headers to response.
We have a query that returns around 1500 nodes, so header of response becomes too large with data that we never use.
NodeJs when fetching pages has a limitation of headers to 8Kb due to some vulnerabilities ([link](https://github.com/nodejs/node/issues/24693#issuecomment-442510749)).
So now we are having `Parse Error: Header overflow` after your plugin update.
For now there is a possibility to change this limitations with extra flag, but would be nice just to disable QueryAnalyzer.

### What is your proposed solution?

Add a checkbox to disable QueryAnalyzer in settings.

**Comment by Kikky** [comment]
We have the same problem

**Comment by nminchow** [comment]
As mentioned [here](https://github.com/wp-graphql/wp-graphql/issues/2548), this issue is also impacting the [gatsby-source-wordpress plugin](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-source-wordpress).

**Comment by justlevine** [comment]
As a temporary workaround, you can use the following snippet to disable the query analyzer programmatically:

```php
add_action( 'graphql_should_analyze_queries', '__return_false' );
```

**Comment by jasonbahl** [comment]
I think short-term I'd recommend doing what @justlevine is suggesting. 

I'll look into some other options here. 

It sounds like 8k is a common limit, so I'm thinking we can implement a default limit to ensure the headers remain within that limit. 

That limit can be filterable for sites that want it to be bigger or smaller. 

Any keys beyond that limit can be chunked into some "Ignored Keys" headers so those looking to make use of the keys (caching clients, etc) can have introspection into this stuff.

**Comment by jasonbahl** [comment]
So, I thought this was addressed in v1.11.2. But it's not. I opened another issue: #2554 

One thing to note, as well, is that the NODE max header size can also be increased. 

In my package.json for my NextJS project, I updated my scripts: 

from this:

```json
  "scripts": {
    ...
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    ...
  }
```

to this: 

```json
  "scripts": {
    ...
    "dev": "NODE_OPTIONS='--max-http-header-size=80000' next dev",
    "build": "NODE_OPTIONS='--max-http-header-size=80000' next build",
    "start": "NODE_OPTIONS='--max-http-header-size=80000' next start",
    ...
  }
```

And I was able to send a plethora of headers to Next without issue. 

That said, the plan to address #2554 is to reduce the header size significantly, expose an `X-GraphQL-Truncated-Keys` header, and then expose all truncated keys in the "extensions" body of the request if GRAPHQL_DEBUG or another filter is true. 

This should allow for caching clients to still make use of X-GraphQL-Keys, and allow developers to debug as necessary when certain queries might not be getting tagged in caches as expected.
